### ISSUE TITLE
Circular dependency error not shown when executing production build with source map flag

### ISSUE THREAD

**Issue by FrciSmrci** [angular/angular-cli#17470]
# 🐞 Bug report

### Command (mark with an `x`)
- [ ] new
- [x] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Is this a regression?
I tested this with Angular CLI 7.3.10 version and it worked as expected. 

### Description
Circular dependency error is not shown when building project with --source-map and --prod flag simultaneously.

## 🔬 Minimal Reproduction
1. Create a circular dependency issue
2. Run ng build --prod --source-map

## 🌍 Your Environment
<pre><code>
Angular CLI: 8.3.26
Node: 12.8.1
OS: darwin x64
Angular: 8.2.14
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.803.26
@angular-devkit/build-angular     0.803.23
@angular-devkit/build-optimizer   0.803.23
@angular-devkit/build-webpack     0.803.23
@angular-devkit/core              8.3.26
@angular-devkit/schematics        8.3.26
@angular/cdk                      8.2.3
@angular/cli                      8.3.26
@ngtools/webpack                  8.3.23
@schematics/angular               8.3.26
@schematics/update                0.803.26
rxjs                              6.5.4
typescript                        3.5.3
webpack                           4.39.2
</code></pre>

**Comment by alan-agius4** [comment]
Hi @FrciSmrci, I tried to replicate this using Angular CLI 9 but I was unable. Can you try to reproduce and share a minimal reproduction?

A good way to make a minimal repro is to create a new app via `ng new repro-app` and adding the minimum possible code to show the problem. Then you can push this repository to github and link it here.

This might be related to your directory structure so its really important to get an accurate repro to diagnose this.

Thanks.

**Comment by FrciSmrci** [comment]
Hi @alan-agius4 ,

I created the <code>repro-app</code>, while setting everything up I came on an idea regarding our app structure you mentioned above. 

Some of the singleton services were provided in root while others (old ones) were provided manually through the core module. I changed the implementation so all of them have providedIn root property which fixed the issue and the errors are now displaying correctly. 

When I'll have a bit more time I'm going to research this into detail so I find the reasoning behind it. 

Thank you for all of your help!
